### DISCUSSION TITLE
Tutorial

### DISCUSSION THREAD

**Discussion by mshalaby007** [shazamio/ShazamIO#41] [Q&A]
Hello, can i get a basic demo on how to use this library to identify a song saved on my local hard disk?

**Comment by dotX12** [comment]
@mshalaby007, hello!
https://github.com/dotX12/ShazamIO/blob/master/examples/recognize_song.py

**Comment by sosopoet** [comment]
Hi, just joined GitHub mainly to say **thanks a million for ShazamIO**. I've been using it with great success recently.

I've seen it written that it needs about a ten second .mp3, but I've found 30 seconds better, for the more obscure tunes I sometimes play. Upload file size is usually less than 500kB. I use Linux and I've added a couple of subprocesses to 'recognise_song.py' that use sox to make a 30 second .wav and lame to make an mp3 of it. I called it shazamit.py and it should be obvious I'm not a good (Python) programmer.

It needs to be called like: 'shazamit.py /home/john/longtune.wav 20 30' where '20' is number of seconds into the song to start and '30' is how many seconds to send. I use a messy way to get the artist and song name from the script output (if it's greater than 160 characters. If it's less then Shazam couldn't find a match).

Hoping it helps.

[shazamit.py.zip](https://github.com/dotX12/ShazamIO/files/10149281/shazamit.py.zip)

**Comment by moshe-solomon** [comment]
hi ,is there a option to get my track list from my shazam user ?

**Reply by dotX12** [reply]
> hi ,is there a option to get my track list from my shazam user ?

Hello, so far there was no such possibility in the plans, and there is not so much time to figure out how to implement authorization for apple id in shazamio.
